### ISSUE TITLE
Cache not invalidated when file content changes (revive false negative)

### ISSUE THREAD

**Issue by vigo** [golangci/golangci-lint#6466]
## Version

```
golangci-lint has version 2.11.4 built with go1.26.1 from 8f3b0c7 on 2026-03-22T17:29:46Z
```

## Description

When a file is modified in a way that introduces a new linter issue (e.g., a previously used parameter becomes unused after a code change), `golangci-lint run` does not detect the issue until the cache is manually cleaned with `golangci-lint cache clean`.

## Steps to reproduce

1. Have a Go function with a used parameter:
```go
func (c *Console) Notify(_ context.Context, webhookURL, message string) error {
    io.WriteString(c.writer, "Webhook: "+webhookURL+"\n")
    // ...
}
```
2. Run `golangci-lint run` — passes (0 issues), as expected.
3. Modify the function so the parameter is no longer used:
```go
func (c *Console) Notify(_ context.Context, webhookURL, message string) error {
    io.WriteString(c.writer, "Webhook: [REDACTED]\n")  // webhookURL no longer referenced
    // ...
}
```
4. Run `golangci-lint run` again — still reports **0 issues** (wrong).
5. Run `golangci-lint cache clean && golangci-lint run` — now correctly reports:
```
unused-parameter: parameter 'webhookURL' seems to be unused, consider removing or renaming it as _ (revive)
```

## Expected behavior

`golangci-lint run` should detect that the file has changed and invalidate the cached result, reporting the `unused-parameter` issue without needing a manual `golangci-lint cache clean`.

## Actual behavior

The cached result from the previous run (0 issues) is returned even though the file content has changed and a new issue should be reported.

## Configuration

```yaml
version: "2"

run:
  timeout: 5m
  tests: false

linters:
  enable:
    - revive
    # ... other linters
  settings:
    revive:
      enable-all-rules: true
```

## Environment

- OS: macOS (Darwin 25.3.0, arm64)
- Go: 1.26.1
- golangci-lint: 2.11.4

## Additional context

This was caught on CI (GitHub Actions) where the cache was different, but consistently missed locally. The pre-commit hook using `golangci-lint-mod` also missed it for the same reason.
